src/main/java/hudson/plugins/analysis/core/BuildHistory.java

```diff
@@ -187,11 +190,11 @@ private boolean isSuccessfulAction(final ResultAction<? extends BuildResult> act
         return null;
     }
 
-    private boolean hasValidResult(final AbstractBuild<?, ?> build) {
+    private boolean hasValidResult(final Run<?, ?> build) {
```

Comment:
Yes, because an ```AbstractBuild``` is a ```Run```. This is a generalization, so there are no backward compatibility issues.